include/cpuinfo.h

```diff
@@ -18,15 +18,15 @@
 #define CPUINFO_ARCH_X86 1
 #endif
 
-#if defined(__x86_64__) || defined(__x86_64) || defined(_M_X64) || defined(_M_AMD64)
+#if defined(__x86_64__) || defined(__x86_64) || (defined(_M_X64) && !defined(_M_ARM64EC)) || (defined(_M_AMD64) && !defined(_M_ARM64EC))
```

Comment:
Defining `CPUINFO_ARCH_X86_64` for `_M_ARM64EC` feels wrong, please explain why this is needed

Reply:
_M_X64 Defined as the integer literal value 100 for compilations that target x64 processors or ARM64EC. So _M_X64  could be 1 when the CPU is actually an ARM CPU.  
Here the condition is `defined(_M_X64) && !defined(_M_ARM64EC))` . When _M_ARM64EC is 1, the expression is false.

Reply:
> It feels wrong to define CPUINFO_ARCH_X86_64 on ARM platform, 

It is the problem I am facing and fixing. _M_X64 is defined when the ABI is ARM64EC. So, without this change, CPUINFO_ARCH_X86_64  is defined when the target ABI is  ARM64EC which is target for ARM64 CPUs.